### PR TITLE
Show (with value hidden) the auth environment variable in system details.

### LINKED ISSUE #914: The presence of CHARMCRAFT_AUTH should be logged
Currently we're logging all CHARMCRAFT_* environment variables, but specifically excluding `CHARMCRAFT_AUTH` to not leak secrets.

However, it's very useful, for debugging purposes, know that `CHARMCRAFT_AUTH` is present or not, so we should log it with a different value, something like...

```
System details: (...); Environment: DESKTOP_SESSION='plasma', CHARMCRAFT_AUTH='<hidden>', ...
```

### PR DESCRIPTION
**Pull request by facundobatista**
Fixes #914.

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -334,8 +334,8 @@ def test_systemdetails_charmcraft_environment():
     )
 
 
-def test_systemdetails_no_auth():
-    """System details specifically excluding secrets."""
+def test_systemdetails_hidden_auth():
+    """System details specifically hidding secrets."""
```
Comment (mr-cal) [inline tests/test_main.py:338]: same as above

File: charmcraft/main.py [hunk 2]
```diff
@@ -102,16 +102,20 @@
 
 def _get_system_details():
     """Produce details about the system."""
-    # prepare the useful environment variables: all CHARMCRAFT* (except AUTH keys)
+    # prepare the useful environment variables: all CHARMCRAFT* (hidding AUTH keys)
```
Comment (mr-cal) [inline charmcraft/main.py:105]: `hidding` -> `hiding`

**Review by mr-cal — APPROVED** [review]
Looks great!

File: charmcraft/main.py (hunk 2 above)
Comment (facundobatista) [reply charmcraft/main.py]: Oh, thanks!

**Pull request merged**